src/polyfem/assembler/GenericProblem.hpp

```diff
@@ -45,6 +45,43 @@ namespace polyfem
 			double form_;
 		};
 
+		class PiecewiseConstantInterpolation : public Interpolation
```

Comment:
These are all pretty much the same class with a different eval function. I suggest creating a parent class 
```c++
class PiecewiseInterpolation : public Interpolation
{
		public:
			void init(const json &params) override;

		private:
			std::vector<double> points_;
			std::vector<double> values_;
			int extend_ = -1; //0-constant, 1-extrapolate, 2-repeat, 3-repeat_offset
			double offset_ = 0.0;
}
```
and then inherit from this class and override eval.